### PR TITLE
wasm3: init at 0.5.0

### PR DESCRIPTION
**Pull request by malbarbo**
###### Motivation for this change

A fast wasm interpreter.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127687` at efb3794a run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/ca39de663090d6be6b4c125dd535b808">wasm3</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/interpreters/wasm3/default.nix:19:3:
```
   |
19 |   installPhase = "mkdir -p $out/bin/; cp -a wasm3 $out/bin/";
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/interpreters/wasm3/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sha256 = "07zzmk776j8ydyxhrnnjiscbhhmz182a62r6aix6kfk5kq2cwia2";
+  };
+
+  # https://github.com/vshymanskyy/uvwasi is fetched by the build script.
+  GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
+
+  nativeBuildInputs = [ cmake git ];
+
+  installPhase = "mkdir -p $out/bin/; cp -a wasm3 $out/bin/";
+
```
Comment (musfay) [inline pkgs/development/interpreters/wasm3/default.nix:20]:
```suggestion
  installPhase = ''
    runHook preInstall
    install -Dm755 wasm3 -t $out/bin
    runHook postInstall
  '';
```

**Review by musfay — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/wasm3/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "wasm3";
+    repo = "wasm3";
+    rev = "v${version}";
+    sha256 = "07zzmk776j8ydyxhrnnjiscbhhmz182a62r6aix6kfk5kq2cwia2";
+  };
+
+  # https://github.com/vshymanskyy/uvwasi is fetched by the build script.
```
Comment (musfay) [inline pkgs/development/interpreters/wasm3/default.nix:14]: The `uvwasi` source should be declared. We can disable it with `-DBUILD_WASI=simple` flag. `nodejs` already bundles it.

**Review by musfay — CHANGES_REQUESTED** [review]

File: pkgs/development/interpreters/wasm3/default.nix (hunk 1 above)
Comment (malbarbo) [reply pkgs/development/interpreters/wasm3/default.nix]: :+1:

File: pkgs/development/interpreters/wasm3/default.nix (hunk 2 above)
Comment (malbarbo) [reply pkgs/development/interpreters/wasm3/default.nix]: For while I just added `-DBUILD_WASI=simple`. In the future we can add an option to build with `uvwasi`.

**Review by musfay — APPROVED** [review]

**Pull request merged**